Feature Text To Speech

Joe91 opened the pull request:
Feature to play around with TTS a little and command-execution on the control-site of the webserver.

- TTS as new playmode (Text from path-line in Webserver)
![grafik](https://user-images.githubusercontent.com/6923271/218837113-445e7014-0a4c-4a47-a1c4-2c91f9b8bf23.png)
- TTS as Command with predefined text in Webserver (Settings-Page)
![grafik](https://user-images.githubusercontent.com/6923271/218836919-c8cc32b9-d974-49e7-8ae5-8670e0492987.png)
- Commands on Control-Page of Webserver:
![grafik](https://user-images.githubusercontent.com/6923271/218648166-d7ede8e0-aa75-4eee-bb47-be5790d30724.png)


TODOs:
- [x] Cleanup Webserver
- [x] Valid TTS-Visuals when using TTS from RFID-Tag
- [x] Implement into new webserver

Joe91 committed:
Merge pull request #1 from biologist79/master

update master

Joe91 committed:
Merge pull request #2 from biologist79/master

merge back master

Joe91 committed:
feature tts as playmode and button-action

Joe91 committed:
fix English translation

fschrempf commented on html/management_EN.html:
```diff
@@ -230,33 +230,33 @@
+					<select class="form-control" id="modId" name="modId">
+						<option class="placeholder" disabled selected value="">Select modification</option>
+						<option value="100">Keylock</option>
+						<option value="179">Schlafe immediately</option>
```
Sleep

fschrempf commented on html/management_EN.html:
```diff
@@ -230,33 +230,33 @@
+						<option value="107">Sleep after end of five tracks</option>
+						<option value="110">Loop playlist</option>
+						<option value="111">Loop track</option>
+						<option value="120">Dimm LEDs (nightmode)</option>
```
Dim

fschrempf commented on html/management_EN.html:
```diff
@@ -230,33 +230,33 @@
+						<option value="120">Dimm LEDs (nightmode)</option>
+						<option value="130">Toggle WiFi</option>
+						<option value="140">Toggle Bluetooth</option>
+						<option value="141">Bluetooth Headphoe aktivation / deaktivation </option>
```
- Missing 'n' in "Headphone"
- 'c' instead of 'k' in "activation/deactivation"
- Maybe "enable/disable" would be even better

fschrempf commented on html/management_EN.html:
```diff
@@ -350,6 +350,7 @@
 								<option value="120">Dimm LEDs (nightmode)</option>
 								<option value="130">Toggle WiFi</option>
 								<option value="140">Toggle Bluetooth</option>
+								<option value="141">Bluetooth Headphoe aktivation / deaktivation </option>
```
See above

Joe91 committed:
some cleanup on the website

Joe91 committed:
english translation review

Joe91 committed:
Merge pull request #3 from biologist79/master

update from origin master

Joe91 committed:
Merge pull request #4 from biologist79/master

Add support for i18next localisation library (#194)

Joe91 committed:
Merge branch 'master' into feature_tts

# Conflicts:
#	html/management.html
#	html/management_DE.html

Joe91 committed:
removed web-stuff and start to add them again.

Joe91 committed:
try to add command-selection

Joe91 committed:
add custom-text-command again

laszloh committed:
Bugfix for automatic file generation

Fixed some errors in the accesspoint html files, which created
double text entries.

Modifed processHtml.py to automatically generate  the needed files.

Following functions are implemented:
 - listed HTML files will be minified but not compressed
 - Files ending in json will be minified & compressed
 - Every other file will be compressed without further post processing

"Flask_Minify" is used as html  and "json" as json minifier.

dist folder is removed since the source folder is used directly.

Joe91 committed:
Merge pull request #5 from laszloh/i18n_bugfix

Bugfix for automatic file generation

Joe91 committed:
Merge pull request #6 from biologist79/master

update master

Joe91 committed:
Merge remote-tracking branch 'origin/master' into feature_tts

laszloh committed:
bugfix: UTF8 encoding & missing translation

Make python explicitely use utf8 as file encoding.

Add missing translation of file upload progress bar.

Joe91 committed:
Merge pull request #7 from biologist79/master

merge master

Joe91 committed:
execute modification works again

Joe91 committed:
Merge branch 'master' into feature_tts

Joe91 committed:
prepare for custom-text

Joe91 committed:
Merge pull request #8 from laszloh/i18n_bugfix

bugfix: UTF8 encoding & missing translation

Joe91 committed:
add custom-text again

Joe91 committed:
don't require text anymore

Joe91 committed:
Merge pull request #9 from biologist79/master

merge master back

Joe91 committed:
fix assignment in webserver

Joe91 commented:
Seems to work again :-)

Joe91 committed:
Merge pull request #10 from biologist79/master

merge master

Joe91 committed:
Merge remote-tracking branch 'origin/master' into feature_tts

# Conflicts:
#	html/locales/de.json
#	html/locales/en.json
#	html/management.html

Joe91 committed:
fix merge issues

Joe91 commented:
Shall I separate the command-execution on the control-site of the webserver to a separated PR?
That way you can execute every possible modification directly on the webser without the need to create a Tag or a Button-Combination for it.

biologist79 commented:
Will try to test this feature within the upcoming days.
